backstory/src/demo/java/com/google/sps/servlets/GPT2Servlet.java

```diff
@@ -65,8 +66,12 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     }
     String generatedText;
     try {
+      if (URLProvider == null) {
+        URLProvider = new StoryManagerURLProvider();
+      }
       StoryManager storyManager = new StoryManagerImpl(text, DEFAULT_MAX_STORY_LENGTH, DEFAULT_TEMPERATURE);
       generatedText = storyManager.generateText();
+      URLProvider.cycleURL();
```

Comment:
Do you always want to cycle or just when there's an exception?

Reply:
On every occasion for the demo.